src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java

```diff
@@ -5161,7 +5152,7 @@ public static Value well_stocked(
     if (checkQuant < 6) return DataTypes.FALSE_VALUE;
     if (checkPrice < (2 * ItemDatabase.getPriceById(itemID))) return DataTypes.FALSE_VALUE;
     // get some data
-    SortedListModel<PurchaseRequest> results = new SortedListModel<PurchaseRequest>();
+    SortedListModel<PurchaseRequest> results = new SortedListModel<>();
```

Comment:
Why is this a SortedListModel?
MallSearchRequest takes ```List<PurchaseRequest> results;```
It makes a request of KoL with ```this.addFormField("sortresultsby", "price");```

This method "Assumes sorted by price". Yes. that is the contract of MallSearchRequest.
The "results" list is never shown in a GUI element. It need not be a ListModel.

Why not a simple:
```
List<PurchaseRequest> results = new ArrayList<>()
```

Reply:
I'm going to guess that when I wrote this in 2021 MallPurchaseRequest took something else and when the change happened in 2022 no one revisited this code until now.  Similarly, if the sorting contract was called out in purchase request code, I missed it and so called it out since well_stocked's implementation depended upon that.  I can't see any reason not to do the cleanup.